DigitalOffice/Tests/UserServiceTests/CommandsTests/LoginCommandTests.cs

```diff
@@ -0,0 +1,120 @@
+using Moq;
+using FluentValidation;
+using UserService.Models;
+using FluentValidation.Results;
+using UserService.Repositories.Interfaces;
+using UserService.Commands.Interfaces;
+using UserService.Commands;
+using System.Net.WebSockets;
+using System.Security.Cryptography;
+using UserService.Database.Entities;
+
+namespace UserServiceUnitTests.CommandsTests
+{
+    public class LoginCommandTests
+    {
+        private Mock<IValidator<UserCredentials>> validatorMock;
+        private Mock<IUserRepository> repositoryMock;
+        private ILoginUserCommand command;
+
+        [SetUp]
+        public void Initialization()
+        {
+            validatorMock = new Mock<IValidator<UserCredentials>>();
+            repositoryMock = new Mock<IUserRepository>();
+
+            repositoryMock
+                .Setup(x => x.GetUserByEmail(It.IsAny<string>()))
+                .Returns(new User());
+
+            validatorMock
+                .Setup(x => x.Validate(It.IsAny<IValidationContext>()))
+                .Returns(new ValidationResult());
+
+            command = new LoginUserCommand(validatorMock.Object, repositoryMock.Object);
+        }
+
+        [Test]
+        public void ShouldPassWhenPasswordsHashMatch()
+        {
+            var shaM = new SHA512Managed();
+            string password = "Example";
+
+            repositoryMock
+                .Setup(x => x.GetUserByEmail(It.IsAny<string>()))
+                .Returns(new User
+                {
+                    PasswordHash = Encoding.UTF8.GetString(
+                        shaM.ComputeHash(
+                        Encoding.UTF8.GetBytes(password)
+                ))
+                });
+
+            var creds = new UserCredentials
+            {
+                Password = password
+            };
+
+            command.Execute(creds);
+        }
+
+        [Test]
+        public void ShouldThrowExceptionWhenPasswordsHashDoesNotMatch()
+        {
+            var shaM = new SHA512Managed();
+            string password = "Example";
+
+            repositoryMock
+                .Setup(x => x.GetUserByEmail(It.IsAny<string>()))
+                .Returns(new User
+                {
+                    PasswordHash = Encoding.UTF8.GetString(
+                        shaM.ComputeHash(
+                        Encoding.UTF8.GetBytes(password)
+                ))
+                });
+
+            var creds = new UserCredentials
+            {
+                Password = "elpmaxE"
+            };
+
+            Assert.Throws<Exception>(() => command.Execute(creds), "Wrong password.");
+        }
+
+        [Test]
+        public void ShouldThrowExceptionWhenUserNotFound()
+        {
+            repositoryMock
+                .Setup(x => x.GetUserByEmail(It.IsAny<string>()))
+                .Returns(null as User);
+
+            var creds = new UserCredentials();
+
+            Assert.Throws<Exception>(() => command.Execute(creds), "User not found.");
+
```

Comment:
Remove it.

Reply:
Do you mean the blank line? If so, removed.

Reply:
Yep